testctrl/docs/specs/container_image_names.md

```diff
@@ -0,0 +1,119 @@
+# Container Image Names
+
+The benchmarking framework uses docker containers for each of the test
+components. Since images are built and pushed to a docker registry often, the
+number of images can grow large. Therefore, this spec defines a naming
+convention that can aid automated clean-up scripts.
+
+## Background
+
+Docker images are encapsulated in repositories in a registry. The naming is a
+bit confusing. In terms of the Docker CLI, a tag is `"httpd:v1"` where `"v1"`
+is the version. This differs from the Registry perspective, where `"httpd"` is
+a repository and `"v1"` is a tag.
+
+Docker registries, like Google Container Registry ([gcr.io]) and DockerHub
+([hub.docker.com]), implement a common [Registry API]. However, docker and this
+API do not natively provide batch clean-up actions. Docker also provides no TTL
+on container images.
+
+[gcr.io]: https://gcr.io
+[hub.docker.com]: https://hub.docker.com
+[Registry API]: https://docs.docker.com/registry/spec/api/
+
+One approach for building a TTL on container images would be to use timestamps
+as versions/tags of the images. However, the Registry API cannot query based on
+these. Checking these requires a separate API call for each repository.
+
+Instead of excessive requests, this framework imposes a strict set of naming
+conventions that can aid in automated clean up.
+
+## Specification
+
+The framework allows a repository to contain exactly 1 container image. The
+repository is named as follows:
+
+```
+[ REGISTRY_PREFIX "/" ]
```

Comment:
not sure how the registry API v2 handles the nested repositories, but 
`gcloud container images list` does seem to list the sub-repositories (or images) as if you were browsing a directory structure.

Example:
```
$ gcloud container images list
prints gcr.io/grpc-testing/jtattermusch_benchmarking_test

$ gcloud container images list --repository gcr.io/grpc-testing/jtattermusch_benchmarking_test
prints gcr.io/grpc-testing/jtattermusch_benchmarking_test/20200417

$ gcloud container images list --repository gcr.io/grpc-testing/jtattermusch_benchmarking_test/20200417
prints gcr.io/grpc-testing/jtattermusch_benchmarking_test/20200417/cxx
```
Is that how things will work when using the registry API?
Understanding the exact way the images will be browsed is important for figuring out the easiest way to cleanup old images.

Reply:
Let's move discussion for both here: https://github.com/codeblooded/grpc/pull/73#discussion_r411456726